dxr/app.py

```diff
@@ -114,27 +122,34 @@ def _search_json(query, tree, query_text, is_case_sensitive, offset, limit, conf
             if is_case_sensitive:
                 params['case'] = 'true'
             return jsonify({'redirect': url_for('.browse', _anchor=line, **params)})
+
     try:
-        count_and_results = query.results(offset, limit)
-        # Convert to dicts for ease of manipulation in JS:
-        results = [{'icon': icon,
-                    'path': path,
-                    'lines': [{'line_number': nb, 'line': l} for nb, l in lines],
-                    'is_binary': is_binary}
-                   for icon, path, lines, is_binary in count_and_results['results']]
+        count, results = query.results(offset, limit)
+        term = query.single_term()
+        promoted_count, promoted = 0, []
+        # Only offer promoted results when at the top of the page and we have only a single term.
+        if offset == 0 and term:
```

Comment:
We can find out by doing the first few lines of results(), which would have to happen anyway. http://git.io/vm3Bp
Then we can iterate over and count the number of FILE and LINE filter.domain's.
We could move that outside of the call to results and pass it in instead.